### PR TITLE
feat(con-1226): Enhance Modal to support full width screen and customized background color

### PR DESCRIPTION
**Pull request by cyrusxzw**
# [CON-1226](https://aesoponline.atlassian.net/browse/CON-1226)

## Issue / Feature description
Modal cannot support full screen, and background color is set to light. Only close icon can be changed based off theme color.

## Development tasks summary
- Add an prop `isFullScreen` to trigger expand to full screen. 
- Add background color based off theme color.
- Change default theme color to `light`. So the default modal background color is light and close icon color is dark, which make more sense. 

## Screenshot

https://user-images.githubusercontent.com/29698458/157137171-1dfd5402-6212-4115-a2b4-3c0f11f45c42.mov

### CODE REVIEW

File: src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -21,16 +21,20 @@ const Modal: ModalType = ({
   children,
   className,
   copy,
+  isFullScreen = false,
   isVisible,
   onClose,
   theme,
 }) => {
-  const currentTheme = useThemeContext(theme, 'dark');
+  const currentTheme = useThemeContext(theme, 'light');
```
Comment (MkMan) [inline src/components/Modal/Modal.tsx:29]: Why is the default theme changing?

**Review by MkMan — COMMENTED** [review]
I noticed in the video that when the theme is `dark` the text is dark against a dark background.

File: src/components/Modal/Modal.tsx (hunk 1 above)
Comment (cyrusxzw) [reply src/components/Modal/Modal.tsx:29]: The previous default theme is `dark`, which means close icon is in dark color. But now, I added background color. `dark` means background color is dark, and close icon is `light`. In order to keep background color is `light` and close icon is `dark` for the existing component using Modal, I change the default color to `light`.

File: src/components/Modal/Modal.types.ts [hunk 2]
```diff
@@ -6,7 +6,9 @@ type ModalAndModalBodyProps = {
   aria?: {
     label?: string;
   };
+  backgroundColor?: string;
```
Comment (kevin-ho87) [inline src/components/Modal/Modal.types.ts:9]: is this prop type used at all?

File: src/components/Modal/Modal.tsx (hunk 1 above)
Comment (kevin-ho87) [reply src/components/Modal/Modal.tsx:29]: just make sure the pages currently using the modal still works as intended

File: src/components/Modal/Modal.types.ts (hunk 2 above)
Comment (cyrusxzw) [reply src/components/Modal/Modal.types.ts:9]: Oh, nope, I forgot to remove it. thanks

**Comment by cyrusxzw** [comment]
> 

It doesn't matter, can always control the color of modal content.

File: src/components/Modal/Modal.tsx (hunk 1 above)
Comment (cyrusxzw) [reply src/components/Modal/Modal.tsx:29]: Yea, I found `SavedCreditCardDeleteModal` and `FormModal` used Modal from Gel. They work well.

**Review by MkMan — APPROVED** [review]

**Review by kevin-ho87 — APPROVED** [review]

**Review by hassanraza0054 — APPROVED** [review]

**Pull request merged**